app/controllers/service_provider_controller.rb

```diff
@@ -32,6 +36,12 @@ def authorization_token
   end
 
   def sp_params
-    params.permit(service_provider: {})
+    return {} unless request.headers['Content-Type'] == 'gzip/json'
+
+    body = request.body.read
+
+    return {} unless body.present?
+
+    JSON.parse(Zlib.gunzip(body))
```

Comment:
So in my mind there are 3 cases we want to handle, in the spirit of "be liberal in what you accept, and strict in what you output":
- GZIP'ed JSON params
- regular JSON params
- form encoded params

So seeing this it seems like we just have empty params for anything that's not gzipped?

Is it possible to allow the other formats in addition to the GZIP? Like something like:


```suggestion
    if request.headers['Content-Type'] == 'gzip/json'
      body = request.body.read
      if body.present?
        JSON.parse(Zlib.gunzip(body))
      else
        {}
      end
    elsif params.key?(:service_provider) # fall back to rails processing of params?
      params.permit(service_provider: {})
    end
```

Reply:
yeah i like that! thanks

Reply:
updated in [`ec73ab3` (#9074)](https://github.com/18F/identity-idp/pull/9074/commits/ec73ab3983dbd3a15d881e80118f6aad62fd8dfa)